crates/weaver_resolver/src/attribute.rs

```diff
@@ -80,16 +81,25 @@ impl AttributeCatalog {
                 note,
                 stability,
                 deprecated,
+                prefix,
             } => {
+                let name;
                 let root_attr = self.root_attributes.get(r#ref);
                 if let Some(root_attr) = root_attr {
                     let mut attr_lineage = AttributeLineage::new(&root_attr.group_id);
 
+                    if *prefix {
+                        // depending on the prefix we either create embedded attribute or normal reference
+                        name = format!("{}.{}", group_prefix, r#ref.clone());
```

Comment:
The `clone` of the `ref` here is probably useless because the `format!` macro doesn't consume its parameters.